docs/app/pages/components/components-sections/input-controls/typeahead/snippets/app.ts

```diff
@@ -0,0 +1,45 @@
+import { Component } from '@angular/core';
+import 'chance';
+import { TypeaheadOptionEvent, TypeaheadKeyService } from '@ux-aspects/ux-aspects';
+
+@Component({
+    selector: 'app',
+    templateUrl: 'app.component.html',
+    styleUrls: ['./app.component.css'],
+})
+export class AppComponent {
+
+    values: ReadonlyArray<string> = [];
+
+    dropdownOpen: boolean = false;
+    selectOnEnter: boolean = false;
+    dropDirection: 'up' | 'down' = 'down';
+    selectFirst: boolean = false;
+
+    value: string = '';
+    input: string = '';
+
+    loadOptionsFn = this.loadOptions.bind(this);
+
+    /** Load the options and filter the them */
+    loadOptions(pageNum: number, pageSize: number, filter: string): ReadonlyArray<string> {
+        return this.values.filter(tag => tag.toLowerCase().indexOf(filter.toLowerCase()) !== -1).slice(pageNum * pageSize, (pageNum + 1) * pageSize);
```

Comment:
I also just noticed one original review comment has been missed, could we add a 1 or 2 second delay here to simulate page loading from the server rather than having pages loading immediately. Might also be good to bump the number of items up from 40 to like 200 to give a good preview of what it is like to have multiple pages.